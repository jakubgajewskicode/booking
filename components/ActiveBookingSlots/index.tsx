import { Grid } from '../../utils/grid';
import { Image } from '../../utils/image';
import { ItemContainer } from '../../utils/imageContainer';

import { H2, P } from '../../utils/Typography';
const ActiveBookingSlots = ({ slots, setActiveWorker, setItemId }: any) => {
  return (
    <Grid>
      {slots.map((item: any) => (
        <ItemContainer
          key={item.id}
          onClick={() => {
            setActiveWorker(false);
            setItemId(item.id);
          }}
        >
          <Image src="https://picsum.photos/200" alt={item.id} />
          <H2>{item.localisedTime}</H2>
          <P>{item.price}</P>
        </ItemContainer>
      ))}
    </Grid>
  );
};

export default ActiveBookingSlots;
