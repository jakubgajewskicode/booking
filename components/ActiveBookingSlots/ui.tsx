import styled from 'styled-components';

import { device } from '../../utils/media';

const ItemContainer = styled.div`
  grid-column: span 8;
  text-align: center;
  cursor: pointer;
  transition: opacity 0.7s ease;

  @media ${device.mobileL} {
    grid-column: span 5;
  }

  @media ${device.tablet} {
    grid-column: span 4;
  }

  :hover {
    opacity: 0.5;
  }
`;

export { ItemContainer };
