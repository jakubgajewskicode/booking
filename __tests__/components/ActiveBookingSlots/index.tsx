import { shallow } from 'enzyme';

import ActiveBookingSlots from '../../../components/ActiveBookingSlots';

const data = {
  slots: [],
  setActiveWorker: false,
  setItemId: 3,
};

describe('Page renders', () => {
  test('renders', () => {
    const activeBookingSlots = shallow(
      <ActiveBookingSlots
        slots={data.slots}
        setActiveWorker={data.setActiveWorker}
        setItemId={data.setItemId}
      />
    );
    expect(activeBookingSlots).toMatchSnapshot();
  });
});
