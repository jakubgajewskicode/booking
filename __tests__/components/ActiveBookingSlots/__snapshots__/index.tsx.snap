// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders renders 1`] = `<grid__Grid />`;
